docs/classic-ui/csrf.md

```diff
@@ -1,13 +1,180 @@
+  from plone.protect import protec
+
+  @protect(CheckAuthenticator)
+  def write_to_api_or_service(self):
+      # code here
+      ...
+  ```
+
+### HTTP POST check with `PostOnly`
+
+Checks whether the request is an HTTP POST and raise `Unauthorized` if not.
+This helps to mitigate clicks on malicious links.
+
+Usage example:
+
+```python
+from plone.protect import PostOnly
+from plone.protect import protect
+
+@protect(PostOnly)
+def write_to_api_or_service(self):
+    # code here
+    ...
+```
+
+## How to add a CSRF-Token to a Link or Form
+
+To pass a token you need either to:
+
+- pass an HTTP GET parameter name `_authenticator` with the token as the value,
+- include a form field named `_authenticator` with the token as the value and submit it with the form, or
+- add an HTTP header named `X-CSRF-TOKEN` with the token as the value.
+
+To add a token as an HTTP GET parameter to a link in a template, you can utilize the authenticator view:
+
+```html
+<tal:authenticator tal:define="authenticator context/@@authenticator">
+  <a href="${python:context.absolute_url()}/myprotected_view?_authenticator=${token}" />
+</tal:authenticator>
+```
+
+To add a hidden field with a token to a form in a template, the above view can be used too like this:
+
+```html
+<span tal:replace="structure context/@@authenticator/authenticator"/>
+```
+
+In Python code, a helper function can be used:
+
+```python
+from plone.protect.authenticator import createToken
+
+token = createToken()
+```
+
+To add an authenticator token to an existing URL with query parameters:
+
+```python
+from plone.protect.authenticator import createToken
+from urllib.parse import urlencode
+from urllib.parse import urlparse
+from urllib.parse import urlunparse
+
+# The existing URL that you want to add a query parameter to
+url = f"https://www.example.com?param1=value1"
+
+# Parse the URL into its component parts
+parsed_url = urlparse(url)
+
+# Add the new query parameters to the 'query' component of the URL
+token_query = urlencode({'_authenticator': createToken()})
+new_query = f'{parsed_url.query}&{token_query}'
+
+# Reassemble the URL with the updated query string
+final_url = urlunparse(
+    (
+        parsed_url.scheme,
+        parsed_url.netloc,
+        parsed_url.path,
+        parsed_url.params,
+        new_query,
+        parsed_url.fragment)
+)
+```
+
+
+## How to allow writes in absence of a protecting token
+
+To allow certain objects to be modified and written to the database without protection, follow these steps:
+
+1. Identify the modified object as a single object in the database.
+2. If an attribute of the object is a "persistent" attribute (e.g., a PersistentDict or PersistentList instance, a BTree, or an annotation), use this instead.
+3. Use the `safeWrite` function to mark the object as safe for writing.
+
+Note: This is the preferred method for allowing modification and writing of specific objects to the database.
```

Comment:
````suggestion
```{note}
This is the preferred method for allowing modification and writing of specific objects to the database.
```
````